### ISSUE TITLE
Display money request in LHN, instead of user's email

### ISSUE THREAD

**Issue by isagoico** [Expensify/App#3737]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Log in to e.cash 
2. Navigate to a conversation
3. Send an IOU 
4. Navigate to IOU

## Expected Result:
The message preview in LHN should mirror what is in the message, instead of the email - i.e. "Requested $20.00 from X for Y"

## Actual Result:
Message preview displays the user's email. This also happens when there's a completed payment. 

## Workaround:
None, visual issue. 

## Platform:
Where is this issue occurring?

 Web ✔️ 
 iOS✔️ 
 Android✔️ 
 Desktop App✔️ 
 Mobile Web✔️ 

**Version Number:** 1.0.73-2

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:**

![image](https://user-images.githubusercontent.com/44479856/123168691-9a9bd500-d478-11eb-894f-cd4dd0f97ed2.png)

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

___ 
From @roryabraham https://expensify.slack.com/archives/C01GTK53T8Q/p1624479971030500
> When the newest action on a report is an IOU payment, the LHN just display the user's email instead of a more user-friendly message such as "Amal paid you $x"

**Comment by MelvinBot** [comment]
Triggered auto assignment to @ctkochan22 (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by ctkochan22** [comment]
Pushing N5 and Cashback stuff out. But labels look good.

**Comment by trjExpensify** [comment]
👋 @ctkochan22 after triaging an issue, can you make sure to add the `external` label so that it continues on the journey to being posted to Upwork. If it can't be worked on externally, then the `internal` label can be added. Going to move this along.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @jliexpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by jliexpensify** [comment]
POSTED: https://www.upwork.com/ab/applicants/1407860465960755200/job-details

**Comment by MelvinBot** [comment]
Triggered auto assignment to @AndrewGable (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by dklymenk** [comment]
As far as I can tell the message displayed on LHN right after you send the payment request is correct, however the message displayed after you reload the page is wrong. This means that the issue is in `getSimplifiedReportObject` function, specifically the part where it extracts html from the message, however it seems like BE response doesn't have `message.html` key/value pair for the IOU objects there.

As far as I can tell this string `Requested <amount> from you for <reason>` comes from BE in 2 cases: when fetching for report history (`API.Report_GetHistory` call) and when getting the IOU object from pusher event, but not from this request on line 301 of `Report.js`:
```
API.Get({
        returnValueList: 'reportStuff',
        reportIDList: chatList.join(','),
        shouldLoadOptionalKeys: true,
        includePinnedReports: true,
    })
```

So I think the easiest way for you would be to probably update that endpoint to return that html, since 2 endpoints already return it.

But if that's not possible I can look into ways to compose this sting on FE from what we already have, although I don't think it will look good. Please let me know, if you would be interested in that.

Thanks.

**Comment by Julesssss** [comment]
Good spot @dklymenk, you're right that we should solve this on the backend, so we'll need to put this issue on hold until this is done.

**Comment by Julesssss** [comment]
Internal [backend issue](https://github.com/Expensify/Expensify/issues/168501) for reference

**Comment by jliexpensify** [comment]
@Julesssss just checking - this issue's still on hold right?

**Comment by Julesssss** [comment]
> @Julesssss just checking - this issue's still on hold right?

Yep, I've been looking into the issue but have been OOO until today.

**Comment by Julesssss** [comment]
Working on this today, will update once the PR is in review.

**Comment by mallenexpensify** [comment]
@Julesssss if this is off hold, can you remove `[ON HOLD]` from the title? If it still on hold, please add `Planning` label to help with GH searching by label

**Comment by MelvinBot** [comment]
@AndrewGable, @jliexpensify Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by AndrewGable** [comment]
@MelvinBot - I think this is still on hold

**Comment by Julesssss** [comment]
Internal PR was raised today to resolve this on the backend.

**Comment by MelvinBot** [comment]
@AndrewGable, @jliexpensify it looks like no one is assigned to work on this job.
Please double the price or add a comment stating why the job isn't being doubled.

**Comment by Julesssss** [comment]
Once again Melvin, it's in progress internally.

**Comment by Julesssss** [comment]
The PR changes slow down the API, so further improvements are required.

**Comment by AndrewGable** [comment]
Still on hold from my understanding

**Comment by Julesssss** [comment]
Yeah, on hold while higher priority tasks are worked through.

**Comment by MelvinBot** [comment]
@AndrewGable, @jliexpensify it looks like no one is assigned to work on this job.
Please double the price or add a comment stating why the job isn't being doubled.

**Comment by Julesssss** [comment]
Internal issue is on hold.

**Comment by AndrewGable** [comment]
Still on hold @MelvinBot

**Comment by AndrewGable** [comment]
Going to assign to @Julesssss as he is more up to date with the internal PR than I am

**Comment by Julesssss** [comment]
Ah, my bad I thought I was assigned too. I'll keep this one updated.

**Comment by Julesssss** [comment]
Okay, the backend PR is once again in review.

**Comment by jliexpensify** [comment]
Hi @Julesssss! Just wanted to confirm what was happening with this one? Is it still on hold? Thanks!

**Comment by Julesssss** [comment]
Hey Jason, the PR is merged and we're awaiting deployment.

**Comment by jliexpensify** [comment]
Awesome, should we remove the [ON HOLD] in the title?

**Comment by Julesssss** [comment]
Yep done. The problem is fully solved by the backend PR, so no external help is required here.

**Comment by jliexpensify** [comment]
Sweet, chatted to Jules and we'll close this out as there's an internal issue!

**Comment by dklymenk** [comment]
I would appreciate it if you update this issue once the change is deployed to staging. I have two client side issues blocked by this and would like to resubmit my PR as soon as it as merged.

Thanks in advance.

**Comment by Julesssss** [comment]
@dklymenk I'm happy to report that it was deployed to staging last night

**Comment by dklymenk** [comment]
We are talking about staging.new.expensify.com, right?

I can still reproduce the issue. 
![DeepinScreenshot_select-area_20210914124815](https://user-images.githubusercontent.com/64093836/133235644-4f5d9057-ed35-44ea-a0f3-341b1a4521a4.png)


Send an IOU and refresh the page.

**Comment by dklymenk** [comment]
Sorry, my bad. I forgot that dev, staging and prod front-ends all use the same prod API. :sweat_smile:
